icicle/utils/utils_kernels.cuh

```diff
@@ -34,7 +34,7 @@ namespace utils_internal {
   {
     int tid = blockDim.x * blockIdx.x + threadIdx.x;
     if (tid < n_elements * batch_size) {
-      int scalar_id = tid % n_elements;
+      long scalar_id = tid % n_elements;
```

Comment:
I would be more specific and use uint32_t or uint64_t.

Reply:
Changed to int64_t as we still need negative indices for inverse NTT here